crates/types/src/replicated_loglet/params.rs

```diff
@@ -86,6 +87,19 @@ impl ReplicatedLogletId {
 )]
 pub struct NodeSet(#[serde_as(as = "HashSet<DisplayFromStr>")] HashSet<PlainNodeId>);
 
+impl Display for NodeSet {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "NodeSet[")?;
+        for (i, id) in self.0.iter().enumerate() {
+            if i != 0 {
+                write!(f, ", ")?;
+            }
+            write!(f, "{}", id)?;
+        }
+        write!(f, "]")
+    }
+}
```

Comment:
Maybe we can just drop the prefix `NodeSet` and `Spread` and keep it as a list `[N1, N2, N2]`.